server/src/main/java/org/candlepin/dto/api/v1/EnvironmentTranslator.java

```diff
@@ -0,0 +1,113 @@
+
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.ObjectTranslator;
+import org.candlepin.dto.api.v1.EnvironmentDTO.EnvironmentContentDTO;
+import org.candlepin.model.Content;
+import org.candlepin.model.Environment;
+import org.candlepin.model.EnvironmentContent;
+import org.candlepin.resource.OwnerResource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * The CapabilityTranslator provides translation from ConsumerCapability model objects to
+ * CapabilityDTOs
+ */
+public class EnvironmentTranslator extends TimestampedEntityTranslator<Environment, EnvironmentDTO> {
+    private static Logger log = LoggerFactory.getLogger(EnvironmentTranslator.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EnvironmentDTO translate(Environment source) {
+        return this.translate(null, source);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EnvironmentDTO translate(ModelTranslator translator, Environment source) {
+        return source != null ? this.populate(translator, source, new EnvironmentDTO()) : null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EnvironmentDTO populate(Environment source, EnvironmentDTO dest) {
+        return this.populate(null, source, dest);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EnvironmentDTO populate(ModelTranslator translator, Environment source,
+        EnvironmentDTO dest) {
+
+        dest = super.populate(translator, source, dest);
+
+        dest.setId(source.getId());
+        dest.setName(source.getName());
+        dest.setDescription(source.getDescription());
+
+        if (translator != null) {
+            dest.setOwner(translator.translate(source.getOwner(), OwnerDTO.class));
+
+            Set<EnvironmentContent> envContents = source.getEnvironmentContent();
+            if (envContents != null) {
+                Collection<EnvironmentContent> envContent = source.getEnvironmentContent();
+                dest.setEnvironmentContent(Collections.<EnvironmentContentDTO>emptyList());
+
+                if (envContent != null) {
+                    ObjectTranslator<Content, ContentDTO> contentTranslator = translator
+                        .findTranslatorByClass(Content.class, ContentDTO.class);
+
+                    for (EnvironmentContent ec : envContent) {
+                        if (ec != null) {
+                            ContentDTO dto = contentTranslator.translate(translator, ec.getContent());
+
+                            if (dto != null) {
+                                log.debug("VRITANT {}", dest);
+                                log.debug("VRITANT {}", dto);
+                                log.debug("VRITANT {}", ec.getEnabled());
+                                dest.addContent(dto, ec.getEnabled().booleanValue());
```

Comment:
Looked into this a bit more and it's as you thought: ec.getEnabled is returning null, which is causing the problem. For environment content, we need to accept null as a value, so once EnvironmentContentDTO is updated accordingly, we can drop the .booleanValue from this and everything should be good.